### PR TITLE
AP_Periph: added per-cell voltages to periph battery monitors

### PR DESCRIPTION
**Pull request by tridge**
This also enables I2C based simulated devices in AP_Periph SITL, allowing for SMBus batteries
image
it also enables ADSB in SITL periph, and should allow for most of the other serial based simulated devices

### CODE REVIEW

File: libraries/AP_HAL_SITL/SITL_Periph_State.cpp [hunk 1]
```diff
@@ -285,6 +288,12 @@ SimMCast::SimMCast(const char *frame_str) :
 void SimMCast::update(const struct sitl_input &input)
 {
     multicast_read();
+    update_external_payload(input);
```
Comment (peterbarker) [inline libraries/AP_HAL_SITL/SITL_Periph_State.cpp:291]: I guess this is required?

**Review by peterbarker — APPROVED** [review]
LGTM

File: libraries/AP_HAL_SITL/SITL_Periph_State.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_HAL_SITL/SITL_Periph_State.cpp:297]: yes

File: Tools/AP_Periph/adsb.cpp [hunk 2]
```diff
@@ -92,6 +70,21 @@ void AP_Periph_FW::adsb_update(void)
             }
         }
     }
+
+    const uint32_t now_ms = AP_HAL::millis();
```
Comment (peterbarker) [inline Tools/AP_Periph/adsb.cpp:74]: This needs a comment as to why it's required....
```suggestion
    // sensors such as the uAvionix Ping need to see a heartbeat so they know which system ID they are part of
    const uint32_t now_ms = AP_HAL::millis();
```

File: Tools/AP_Periph/battery.cpp [hunk 3]
```diff
@@ -75,7 +75,53 @@ void AP_Periph_FW::can_battery_update(void)
...
+    auto* pkt = new ardupilot_equipment_power_BatteryInfoAux;
+    uint8_t* buffer = new uint8_t[ARDUPILOT_EQUIPMENT_POWER_BATTERYINFOAUX_MAX_SIZE];
+    if (pkt == nullptr || buffer == nullptr) {
+        delete pkt;
+        delete [] buffer;
+        return;
+    }
+    const auto &cell_voltages = battery_lib.get_cell_voltages(instance);
+			
+    for (uint8_t i = 0; i < AP_BATT_MONITOR_CELLS_MAX; i++) { // AP_BATT_MONITOR_CELLS_MAX might need changing
```
Comment (peterbarker) [inline Tools/AP_Periph/battery.cpp:102]:
```suggestion
    for (uint8_t i = 0; i < ARRAY_SIZE(cell_voltages.cells); i++) {
```

File: libraries/GCS_MAVLink/GCS_MAVLink.cpp [hunk 4]
```diff
@@ -22,7 +22,7 @@ This provides some support code and variables for MAVLink enabled sketches
 
 #include "GCS_config.h"
 
-#if HAL_GCS_ENABLED
+#if HAL_GCS_ENABLED || HAL_MAVLINK_BINDINGS_ENABLED
```
Comment (peterbarker) [inline libraries/GCS_MAVLink/GCS_MAVLink.cpp:25]:
```suggestion
#if HAL_MAVLINK_BINDINGS_ENABLED
```

File: Tools/AP_Periph/battery.cpp (hunk 3 above)
Comment (tridge) [reply Tools/AP_Periph/battery.cpp:102]: done

File: libraries/GCS_MAVLink/GCS_MAVLink.cpp (hunk 4 above)
Comment (tridge) [reply libraries/GCS_MAVLink/GCS_MAVLink.cpp]: done, needed change in HAL_ChibiOS too

File: Tools/AP_Periph/adsb.cpp (hunk 2 above)
Comment (tridge) [reply Tools/AP_Periph/adsb.cpp:74]: done

**Pull request merged**